### PR TITLE
Bug 1827753: autoselect query from metric dropdown

### PR DESCRIPTION
**Pull request by vikram-raj**
**Fixes**: 
https://issues.redhat.com/browse/ODC-3064

**Analysis / Root cause**: 
When clicking on any metrics graph in the dashboard or sidebar in topology view takes you to metrics in the monitoring tab but the query field remains unchanged

**Solution Description**: 
autoselect query from the metric dropdown if it matches.

**Screen shots / Gifs for design review**: 
monitoring

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by vikram-raj** [comment]
/retitle Bug 1827753: autoselect query from metric dropdown

File: frontend/packages/dev-console/src/components/monitoring/queries.ts [hunk 1]
```diff
@@ -17,49 +17,17 @@ export interface MonitoringQuery {
...
-  PODS_BY_CPU: 'CPU Usage',
-  PODS_BY_MEMORY: 'Memory Usage',
-  PODS_BY_FILESYSTEM: 'Filesystem Usage',
-  PODS_BY_NETWORK_IN: 'Receive Bandwidth',
-  PODS_BY_NETWORK_OUT: 'Transmit Bandwidth',
+  'CPU Usage': 'CPU Usage',
+  'Memory Usage': 'Memory Usage',
+  'Filesystem Usage': 'Filesystem Usage',
+  'Receive Bandwidth': 'Receive Bandwidth',
+  'Transmit Bandwidth': 'Transmit Bandwidth',
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/monitoring/queries.ts:24]: In my opinion the Idea of having enum/const variables is that you can change the value (string) in one place and those variables used elsewhere should receive the changed string. But here in this case you need to change both Key  and Value, which is not very useful.

File: frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx [hunk 2]
```diff
@@ -32,8 +37,8 @@ const MetricsQueryInput: React.FC = () => {
     state.UI.getIn(['queryBrowser', 'queries', 0]).toJS(),
   );
   const dispatch = useDispatch();
-
   const [title, setTitle] = React.useState('Select Query');
+  const [selectedKey, setselectedKey] = React.useState('');
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx:41]:
```suggestion
  const [selectedKey, setSelectedKey] = React.useState('');
```

**Comment by karthikjeeyar** [comment]
On clicking CPU usage graph, it takes me to the monitoring page which has list view under the graph but the graph is not showing up. By inspecting the page, graph appears again.
 

graph-appears-on-inspecting

**Comment by vikram-raj** [comment]
> On clicking CPU usage graph, it takes me to the monitoring page which has list view under the graph but the graph is not showing up. By inspecting the page, graph appears again.
> 

The graph takes some time to load.

**Comment by vikram-raj** [comment]
cc: @openshift/team-devconsole-ux

File: frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx [hunk 3]
```diff
@@ -48,13 +53,25 @@ const MetricsQueryInput: React.FC = () => {
...
-    const query = queries?.query;
+    const q = queries?.query;
     const text = queries?.text;
-    if (text && text.localeCompare(query) !== 0) {
+    if (text && text.localeCompare(q) !== 0) {
       setTitle(CUSTOM_QUERY);
       setIsPromQlDisabled(true);
+      if (query) {
+        removeQueryArgument('query0');
+      }
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx:63]: what's the use of this , i see one in onChange

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx:63]: It removes query params from the URL. If the user starts editing the promql or changes the query using the query dropdown.

**Comment by vikram-raj** [comment]
> Quick question on this @vikram-raj
> Our Out of the Box _CPU Usage_ query is per project, correct?
> I'm assuming that's why when clicking on the _CPU Usage_ card from the dashboard, it defaults to _CPU Usage_, but when clicking on _CPU Usage_ from the side panel in topology, it defaults to _Custom Query_

Correct, out of box _CPU Usage_ query is per project. and CPU Usage from the side panel is per workload.

File: frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx (hunk 3 above)
Comment (invincibleJai) [reply frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx:63]: can we handle both scenarios in hook itself

**Comment by karthikjeeyar** [comment]
verified locally and works fine
/lgtm

**Comment by invincibleJai** [comment]
@vikram-raj it works fine in other scenarios however i see issue ns change if came from topology to metrics see GIF attached.

May-07-2020 17-58-50

**Comment by vikram-raj** [comment]
> @vikram-raj it works fine in other scenarios however i see issue ns change if came from topology to metrics see GIF attached.

@invincibleJai We have this issue in master and we can handle it separately. It is happening because the workload query is not under the pre-canned query that we define for the metrics dropdown.

**Pull request merged**